File: anvil/tests/fixtures/cpp/bad_code/syntax_error.cpp
/**
 * C++ file with syntax errors.
 *
 * This fixture is intentionally broken to test syntax error detection.
 */

#include <iostream>
#include <string>

// Missing semicolon after class definition
class BrokenClass {
 public:
    void method() {
        std::cout << "Test" << std::endl;
    }
}  // Missing semicolon here

// Mismatched braces
void broken_function() {
    if (true) {
        std::cout << "Missing closing brace" << std::endl;
    // Missing closing brace
}

// Invalid function declaration
int another_broken(int x, int y {
    return x + y;
}

// Unmatched template brackets
template<typename T
class BrokenTemplate {
    T value;
};

// Missing return type
broken_return() {
    return 42;
}

// Invalid main
int main( {
    std::cout << "Broken main" << std::endl
    return 0;
}

File: anvil/tests/fixtures/python/bad_code/syntax_error.py
"""
Python file with syntax errors.

This fixture is intentionally broken to test syntax error detection.
"""


# Missing colon after function definition
def broken_function()
    return "This won't parse"


# Unmatched parenthesis
def another_broken(arg1, arg2:
    return arg1 + arg2


# Invalid indentation
class BadClass:
def broken_method(self):
        return "Bad indentation"


# Missing closing quote
message = "This string is not closed


# Invalid syntax with if
if True
    print("Missing colon")
